File: fe/mobile/android/app/src/main/java/com/capstone/di/NotificationModule.java
package com.capstone.di;

import com.capstone.data.repository.NotificationRepositoryImpl;
import com.capstone.data.source.remote.NotificationApi;
import com.capstone.domain.repository.NotificationRepository;
import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;
import retrofit2.Retrofit;

import javax.inject.Singleton;

/**
 * Dagger Module cấu hình các dịch vụ thông báo (Notification).
 */
@Module
@InstallIn(SingletonComponent.class)
public class NotificationModule {

    @Provides
    @Singleton
    public NotificationApi provideNotificationApi(Retrofit retrofit) {
        return retrofit.create(NotificationApi.class);
    }

    @Provides
    @Singleton
    public NotificationRepository provideNotificationRepository(NotificationApi api) {
        return new NotificationRepositoryImpl(api);
    }
}

File: fe/mobile/android/app/src/main/java/com/capstone/di/NotificationModule.kt
package com.capstone.di

import com.capstone.data.repository.NotificationRepositoryImpl
import com.capstone.data.source.remote.NotificationApi
import com.capstone.domain.repository.NotificationRepository
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import retrofit2.Retrofit
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
object NotificationModule {

    @Provides
    @Singleton
    fun provideNotificationApi(retrofit: Retrofit): NotificationApi {
        return retrofit.create(NotificationApi::class.java)
    }

    @Provides
    @Singleton
    fun provideNotificationRepository(api: NotificationApi): NotificationRepository {
        return NotificationRepositoryImpl(api)
    }
}
